Add a couple more tests.

frewsxcv opened the pull request.

frewsxcv committed:
Add test for self-signed cert.

frewsxcv committed:
Add test for cert with invalid CN name.

sfackler commented on src/lib.rs:
```diff
@@ -1023,6 +1023,34 @@ mod test {
+    }
+
+    #[test]
+    fn wrong_host_cert() {
```
There's already one of these up above at bad_domain - maybe get rid of that one?

frewsxcv replied:
https://github.com/sfackler/schannel-rs/pull/8/commits/df138911d339fad9a37d1ed3e684af5311dfff17

frewsxcv committed:
Remove `bad_domain` test.

https://github.com/sfackler/schannel-rs/pull/8/files/825f88b1014fcd9c9ee097386fd4e285d4fafcbf#r65108762
